stan/math/prim/mat/fun/MatrixExponential.h

```diff
@@ -15,7 +15,7 @@
 #define STAN_MATH_PRIM_MAT_FUN_MATRIXEXPONENTIAL_H
 
 #include <stan/math/prim/scal/fun/value_of_rec.hpp>
```

Comment:
Change this to include:
`stan/math/prim/mat/fun/value_of_rec.hpp`

Note: you should be including the `mat` version, not the `scal` version.